packages/plugins/Traderxyz/src/apis/nftSwap.tsx

```diff
@@ -0,0 +1,18 @@
+import { NftSwap } from '@traderxyz/nft-swap-sdk'
+import { ExternalProvider, Web3Provider } from '@ethersproject/providers'
+import { useWeb3Provider, ChainId } from '@masknet/web3-shared-evm'
+import { memoize } from 'lodash-unified'
+
+const useTraderApi_ = memoize(
+    (chainId: ChainId) => {
+        const p = useWeb3Provider()
+        const provider = new Web3Provider(p as unknown as ExternalProvider, chainId)
+        const signer = provider.getSigner(0)
+        return new NftSwap(provider, signer, chainId)
+    },
+    (chainId: ChainId) => String(chainId),
+)
+
+export function useTraderApi(chainId: number) {
+    return useTraderApi_(chainId)
+}
```

Comment:
```suggestion
export const useTraderApi = memoize(
    (chainId: ChainId) => {
        const p = useWeb3Provider()
        const provider = new Web3Provider(p as unknown as ExternalProvider, chainId)
        const signer = provider.getSigner(0)
        return new NftSwap(provider, signer, chainId)
    },
    (chainId: ChainId) => String(chainId),
)
```

Reply:
Due to this error i used the the current aproach

<img width="939" alt="image" src="https://user-images.githubusercontent.com/3579069/168495996-b5392c66-2308-4a25-a4b6-576fea844a34.png">